tools/agentic_import/README.md

```diff
@@ -0,0 +1,264 @@
+  - [Directory Structure](#directory-structure)
+  - [Debugging](#debugging)
+    - [Gemini CLI Debugging](#gemini-cli-debugging)
+    - [Log Structure](#log-structure)
+
+## Prerequisites
+
+Before starting the import process, ensure you have the following installed and configured:
+
+### Required Tools
+1. **Gemini CLI**: Install the Gemini command-line interface
+   - Installation instructions vary by platform
+   - Refer to: https://github.com/google-gemini/gemini-cli
+
+2. **Data Commons API Key**: Set up your DC_API_KEY environment variable
+   - Refer to: https://docs.datacommons.org/api/#obtain-an-api-key
+   ```bash
+   export DC_API_KEY="your_data_commons_api_key_here"
+   ```
+
+3. **Python Environment**: Python 3.11+ with required dependencies
+
+## Setup
+
+### Step 1: Repository Setup
+
+Clone the Data Commons data repository and set up environment variables:
+
+```bash
+# Clone the repository
+git clone https://github.com/datacommonsorg/data.git
+export DC_DATA_REPO_PATH="/path/to/your/cloned/data/repo"
+cd $DC_DATA_REPO_PATH
+```
+
+### Step 2: Environment Setup
+
+Set up the Python virtual environment:
+
+```bash
+# Setup Python virtual environment
+./run_tests.sh -r
+
+# Activate the virtual environment
+source .env/bin/activate
+
+# Note: Use this Python environment for all subsequent commands
+# as it has all required dependencies installed
+```
+
+### Step 3: Working Directory Preparation
+
+Create or navigate to your working directory where input data and metadata files will be stored:
+
+```bash
+# Create working directory
+mkdir -p /path/to/working/directory
+cd /path/to/working/directory
+export WORKING_DIR=$(pwd)
+
+# Ensure you have your input data and metadata files in this directory
+# Expected structure:
+# working_directory/
+# ├── input_data.csv
+# ├── metadata_file1.json
+# ├── metadata_file2.xml
+# ├── metadata_file3.txt
+# └── ... (other metadata files)
+```
+
+> **Note**: For downloading SDMX data and metadata files, see the [SDMX CLI documentation](../sdmx_import/README.md).
+
+## Data Import Process
+
+**IMPORTANT:** Run all the following steps from within your working directory with the Python virtual environment activated.
+
+```bash
+# Confirm you are in the working directory
+cd $WORKING_DIR
+
+# Confirm virtual environment is activated
+source $DC_DATA_REPO_PATH/.env/bin/activate
+```
+
+### Step 4: Data Sampling
+
+Create a sample of your data using the data_sampler utility. This helps with initial testing and validation:
+
+```bash
+# Sample your input CSV data (maximum 30 lines)
+python $DC_DATA_REPO_PATH/tools/statvar_importer/data_sampler.py \
+  --sampler_input="$WORKING_DIR/input_data.csv" \
+  --sampler_output="$WORKING_DIR/sample_data.csv" \
+  --max_rows=30
+```
+
+**Parameters:**
+- `--sampler_input`: Path to your original CSV data file
+- `--sampler_output`: Path where the sample will be saved
+- `--max_rows`: Maximum number of rows to sample (recommended: 30 for testing)
```

Comment:
Should add a note to set this large enough to get all unique values for all columns for non-SDMX sources
which don't have metadata listing all column values.